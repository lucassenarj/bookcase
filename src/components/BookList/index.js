import React from "react";
import PropTypes from "prop-types";
import BookItem from "./../BookItem";
import "./style.css";

function BookList({ books }) {
  return (
    <div className="row justify-content-center">
      {
        books.map(({ volumeInfo }, index) => <BookItem book={volumeInfo} key={index} />)
      }
    </div>
  );
}
BookList.propTypes = {
  books: PropTypes.arrayOf(PropTypes.object).isRequired,
}

export default BookList;
