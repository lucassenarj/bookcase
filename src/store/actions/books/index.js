import api from "services/api";

export const BOOKS_GET_LIST_ERROR = "BOOKS_GET_LIST_ERROR";
export const BOOKS_GET_LIST_REQUEST = "BOOKS_GET_LIST_REQUEST";
export const BOOKS_GET_LIST_SUCCESS = "BOOKS_GET_LIST_SUCCESS";

export function getBooks(terms) {
  return async (dispatch) => {
    try {
      dispatch({
        type: BOOKS_GET_LIST_REQUEST
      });

      const { data } = await api.get(`/volumes?q=${terms}&orderBy=relevance&maxResults=36`);
      
      await dispatch({
        type: BOOKS_GET_LIST_SUCCESS,
        payload: data.items,
      });

    } catch (error) {
      dispatch({
        type: BOOKS_GET_LIST_ERROR,
        payload: error.getMessage(),
      });
    }
  }
}
