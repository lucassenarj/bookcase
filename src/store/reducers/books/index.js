import {
  BOOKS_GET_LIST_ERROR,
  BOOKS_GET_LIST_REQUEST,
  BOOKS_GET_LIST_SUCCESS,
} from "store/actions";

const initialState = {
  books: [],
  loading: true,
  error: false,
};

const app = (state = initialState, action) => {
  switch(action.type) {
    case BOOKS_GET_LIST_REQUEST: {
      return {
        ...state,
        loading: true,
        error: false,
        books: [],
      };
    }
    case BOOKS_GET_LIST_ERROR: {
      return {
        ...state,
        error: true,
        loading: false,
        books: [],
      };
    }
    case BOOKS_GET_LIST_SUCCESS: {
      const books = action.payload;
      return {
        ...state,
        loading: false,
        error: false,
        books,
      };
    }
    default: 
      return state;
  }
};

export default app;
