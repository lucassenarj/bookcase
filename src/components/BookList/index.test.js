import React from "react";
import { render } from "@testing-library/react";
import BookList from "./";
import books from "./../../utils/mocks/bookList";


describe("BookList Component", () => {
  it("renders correctly", () => {
    const { container } = render(<BookList books={books.items} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});