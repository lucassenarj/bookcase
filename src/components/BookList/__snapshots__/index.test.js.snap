// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList Component renders correctly 1`] = `
<div
  class="row justify-content-center"
>
  <div
    class="col-lg-4 col-md-6 col-sm-8 btn-click"
    data-testid="btn-click"
  >
    <div
      class="single-blog mt-30 wow fadeInUpBig"
      data-wow-delay="0.4s"
      data-wow-duration="1s"
    >
      <div
        class="blog-image"
      >
        <div>
          <img
            alt="news"
            src="http://books.google.com/books/content?id=DKcWE3WXoj8C&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
          />
        </div>
      </div>
      <div
        class="blog-content"
      >
        <h4
          class="blog-title"
        >
          <div>
            Harry Potter and International Relations
          </div>
        </h4>
        <div
          class="blog-author d-flex align-items-center"
        >
          <div
            class="author-image"
          >
            <img
              alt="author"
              src="author-1.jpg"
            />
          </div>
          <div
            class="author-content media-body"
          >
            <h6
              class="sub-title"
            >
              Daniel H. Nexon
              Iver B. Neumann
            </h6>
            <p
              class="text"
            >
              2006
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="col-lg-4 col-md-6 col-sm-8 btn-click"
    data-testid="btn-click"
  >
    <div
      class="single-blog mt-30 wow fadeInUpBig"
      data-wow-delay="0.4s"
      data-wow-duration="1s"
    >
      <div
        class="blog-image"
      >
        <div>
          <img
            alt="news"
            src="http://books.google.com/books/content?id=2sSMCwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
          />
        </div>
      </div>
      <div
        class="blog-content"
      >
        <h4
          class="blog-title"
        >
          <div>
            Harry Potter and the Cursed Child â€“ Parts One and Two (Special Rehearsal Edition)
          </div>
        </h4>
        <div
          class="blog-author d-flex align-items-center"
        >
          <div
            class="author-image"
          >
            <img
              alt="author"
              src="author-1.jpg"
            />
          </div>
          <div
            class="author-content media-body"
          >
            <h6
              class="sub-title"
            >
              J.K. Rowling
              John Tiffany
              Jack Thorne
            </h6>
            <p
              class="text"
            >
              2016-07-31
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="col-lg-4 col-md-6 col-sm-8 btn-click"
    data-testid="btn-click"
  >
    <div
      class="single-blog mt-30 wow fadeInUpBig"
      data-wow-delay="0.4s"
      data-wow-duration="1s"
    >
      <div
        class="blog-image"
      >
        <div>
          <img
            alt="news"
            src="http://books.google.com/books/content?id=FCPdzQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
          />
        </div>
      </div>
      <div
        class="blog-content"
      >
        <h4
          class="blog-title"
        >
          <div>
            Harry Potter Cookbook
          </div>
        </h4>
        <div
          class="blog-author d-flex align-items-center"
        >
          <div
            class="author-image"
          >
            <img
              alt="author"
              src="author-1.jpg"
            />
          </div>
          <div
            class="author-content media-body"
          >
            <h6
              class="sub-title"
            >
              Alvin Brewster
            </h6>
            <p
              class="text"
            >
              2020-06-25
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="col-lg-4 col-md-6 col-sm-8 btn-click"
    data-testid="btn-click"
  >
    <div
      class="single-blog mt-30 wow fadeInUpBig"
      data-wow-delay="0.4s"
      data-wow-duration="1s"
    >
      <div
        class="blog-image"
      >
        <div>
          <img
            alt="news"
            src="http://books.google.com/books/content?id=iO5pApw2JycC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
          />
        </div>
      </div>
      <div
        class="blog-content"
      >
        <h4
          class="blog-title"
        >
          <div>
            The Ivory Tower and Harry Potter
          </div>
        </h4>
        <div
          class="blog-author d-flex align-items-center"
        >
          <div
            class="author-image"
          >
            <img
              alt="author"
              src="author-1.jpg"
            />
          </div>
          <div
            class="author-content media-body"
          >
            <h6
              class="sub-title"
            >
              Lana A. Whited
            </h6>
            <p
              class="text"
            >
              2004
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="col-lg-4 col-md-6 col-sm-8 btn-click"
    data-testid="btn-click"
  >
    <div
      class="single-blog mt-30 wow fadeInUpBig"
      data-wow-delay="0.4s"
      data-wow-duration="1s"
    >
      <div
        class="blog-image"
      >
        <div>
          <img
            alt="news"
            src="http://books.google.com/books/content?id=BoX-6R21MgQC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
          />
        </div>
      </div>
      <div
        class="blog-content"
      >
        <h4
          class="blog-title"
        >
          <div>
            The Psychology of Harry Potter
          </div>
        </h4>
        <div
          class="blog-author d-flex align-items-center"
        >
          <div
            class="author-image"
          >
            <img
              alt="author"
              src="author-1.jpg"
            />
          </div>
          <div
            class="author-content media-body"
          >
            <h6
              class="sub-title"
            >
              Neil Mulholland
            </h6>
            <p
              class="text"
            >
              2009-06-22
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="col-lg-4 col-md-6 col-sm-8 btn-click"
    data-testid="btn-click"
  >
    <div
      class="single-blog mt-30 wow fadeInUpBig"
      data-wow-delay="0.4s"
      data-wow-duration="1s"
    >
      <div
        class="blog-image"
      >
        <div>
          <img
            alt="news"
            src="http://books.google.com/books/content?id=1WgGDAAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
          />
        </div>
      </div>
      <div
        class="blog-content"
      >
        <h4
          class="blog-title"
        >
          <div>
            Heroism in the Harry Potter Series
          </div>
        </h4>
        <div
          class="blog-author d-flex align-items-center"
        >
          <div
            class="author-image"
          >
            <img
              alt="author"
              src="author-1.jpg"
            />
          </div>
          <div
            class="author-content media-body"
          >
            <h6
              class="sub-title"
            >
              Katrin Berndt
              Lena Steveker
            </h6>
            <p
              class="text"
            >
              2016-04-22
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="col-lg-4 col-md-6 col-sm-8 btn-click"
    data-testid="btn-click"
  >
    <div
      class="single-blog mt-30 wow fadeInUpBig"
      data-wow-delay="0.4s"
      data-wow-duration="1s"
    >
      <div
        class="blog-image"
      >
        <div>
          <img
            alt="news"
            src="http://books.google.com/books/content?id=Aaug_RnI-xQC&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
          />
        </div>
      </div>
      <div
        class="blog-content"
      >
        <h4
          class="blog-title"
        >
          <div>
            The Irresistible Rise of Harry Potter
          </div>
        </h4>
        <div
          class="blog-author d-flex align-items-center"
        >
          <div
            class="author-image"
          >
            <img
              alt="author"
              src="author-1.jpg"
            />
          </div>
          <div
            class="author-content media-body"
          >
            <h6
              class="sub-title"
            >
              Andrew Blake
            </h6>
            <p
              class="text"
            >
              2002
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="col-lg-4 col-md-6 col-sm-8 btn-click"
    data-testid="btn-click"
  >
    <div
      class="single-blog mt-30 wow fadeInUpBig"
      data-wow-delay="0.4s"
      data-wow-duration="1s"
    >
      <div
        class="blog-image"
      >
        <div>
          <img
            alt="news"
            src="http://books.google.com/books/content?id=3PGBUrScs-YC&printsec=frontcover&img=1&zoom=1&source=gbs_api"
          />
        </div>
      </div>
      <div
        class="blog-content"
      >
        <h4
          class="blog-title"
        >
          <div>
            Harry Potter and the Bible
          </div>
        </h4>
        <div
          class="blog-author d-flex align-items-center"
        >
          <div
            class="author-image"
          >
            <img
              alt="author"
              src="author-1.jpg"
            />
          </div>
          <div
            class="author-content media-body"
          >
            <h6
              class="sub-title"
            >
              Richard Abanes
            </h6>
            <p
              class="text"
            >
              2001
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="col-lg-4 col-md-6 col-sm-8 btn-click"
    data-testid="btn-click"
  >
    <div
      class="single-blog mt-30 wow fadeInUpBig"
      data-wow-delay="0.4s"
      data-wow-duration="1s"
    >
      <div
        class="blog-image"
      >
        <div>
          <img
            alt="news"
            src="http://books.google.com/books/content?id=YhnqzQEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"
          />
        </div>
      </div>
      <div
        class="blog-content"
      >
        <h4
          class="blog-title"
        >
          <div>
            Unofficial Harry Potter Cookbook
          </div>
        </h4>
        <div
          class="blog-author d-flex align-items-center"
        >
          <div
            class="author-image"
          >
            <img
              alt="author"
              src="author-1.jpg"
            />
          </div>
          <div
            class="author-content media-body"
          >
            <h6
              class="sub-title"
            >
              Meghan Gilb
            </h6>
            <p
              class="text"
            >
              2020-09-24
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="col-lg-4 col-md-6 col-sm-8 btn-click"
    data-testid="btn-click"
  >
    <div
      class="single-blog mt-30 wow fadeInUpBig"
      data-wow-delay="0.4s"
      data-wow-duration="1s"
    >
      <div
        class="blog-image"
      >
        <div>
          <img
            alt="news"
            src="http://books.google.com/books/content?id=OwOdDwAAQBAJ&printsec=frontcover&img=1&zoom=1&edge=curl&source=gbs_api"
          />
        </div>
      </div>
      <div
        class="blog-content"
      >
        <h4
          class="blog-title"
        >
          <div>
            Estudo Comparativo Da VersÃƒÂ£o De Harry Potter Y La Piedra Filosofal Em RelaÃƒÂ§ÃƒÂ£o ÃƒÂ  Narrativa Da Obra Original
          </div>
        </h4>
        <div
          class="blog-author d-flex align-items-center"
        >
          <div
            class="author-image"
          >
            <img
              alt="author"
              src="author-1.jpg"
            />
          </div>
          <div
            class="author-content media-body"
          >
            <h6
              class="sub-title"
            >
              Adriane Viz Veiga
            </h6>
            <p
              class="text"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
